pkg/workload/tpcc/tpcc.go

```diff
@@ -137,6 +141,61 @@ func (w *waitSetter) String() string {
 	}
 }
 
+var _ pgx.QueryTracer = fileLoggerQueryTracer{}
+
+type fileLoggerQueryTracer struct {
+	filePath string
+	file     *os.File
+}
+
+func newFileLoggerQueryTracer(filePath string) (*fileLoggerQueryTracer, error) {
+	tracer := &fileLoggerQueryTracer{filePath: filePath}
+	file, err := os.OpenFile(tracer.filePath, os.O_CREATE|os.O_TRUNC|os.O_RDWR, 0666)
+	if err != nil {
+		return nil, err
+	}
+	tracer.file = file
+	return tracer, nil
+}
+
+func (t *fileLoggerQueryTracer) Close() {
+	if t.file != nil {
+		t.file.Close()
+		t.file = nil
+	}
+}
+
+func (t fileLoggerQueryTracer) Write(p []byte) (n int, err error) {
+	if t.file == nil {
+		return 0, errors.New("file not initialized")
+	}
+	return t.file.Write(p)
+}
+
+func (t fileLoggerQueryTracer) TraceQueryStart(
+	ctx context.Context, conn *pgx.Conn, data pgx.TraceQueryStartData,
+) context.Context {
+	if t.file == nil {
+		panic("file not initialized")
+	}
+	whitespacePattern := regexp.MustCompile(`\s+`)
+	fmt.Fprintf(t, "Query: %s\n", whitespacePattern.ReplaceAllString(strings.TrimSpace(data.SQL), " "))
+	return ctx
+}
+
+func (t fileLoggerQueryTracer) TraceQueryEnd(
+	ctx context.Context, conn *pgx.Conn, data pgx.TraceQueryEndData,
+) {
+	if t.file == nil {
+		panic("file not initialized")
+	}
+	if data.Err != nil {
```

Comment:
Nit: IMO, this branch is more readable as `if data.Err == nil { ... } else { .. }` to avoid the `not not nil` on the `else` branch.